pkg/runtest/run.go

```diff
@@ -148,8 +138,8 @@ func (ctx *Context) Run() error {
 		if !verbose || ctx.Verbose {
 			ctx.log("%-38v: %v", req.name, result)
 		}
-		if req.Bin != "" {
-			os.Remove(req.Bin)
+		if req.Request != nil && req.Request.BinaryFile != "" {
```

Comment:
Can req.Request ever be nil? When?

s/req.Request.BinaryFile/req.BinaryFile/ (you use the short form on the next line anyway)

Reply:
> Can req.Request ever be nil? 

It can. In some places we create fake RunRequests just to signalize that it was broken/skipped:

https://github.com/google/syzkaller/blob/20bf80e1d6eb340a35a841a099b8dca81e42ff5e/pkg/runtest/run.go#L270-L273